src/plugins/mer/merbuildconfiguration.cpp

```diff
@@ -89,9 +90,15 @@ MerBuildConfiguration::MerBuildConfiguration(Target *target, Core::Id id)
         if (!isReallyActive())
             return;
         QTC_ASSERT(project(), return);
-        if (!project()->files(Project::AllFiles).contains(document->filePath())
-                || !document->filePath().toString().contains(QRegExp("\\.spec$|\\.yaml$")))
+        if (!document->filePath().toString().contains(QRegExp("\\.spec$|\\.yaml$")))
             return;
+        if (!project()->files(Project::AllFiles).contains(document->filePath())) {
+            if (document->filePath().isChildOf(project()->rootProjectDirectory())) {
+                MessageManager::write(tr("Warning: RPM *.spec (or *.yaml) file not registered to the project."),
```

Comment:
I don't understand, in which situation do we end up in here?

Reply:
When the .spec/.yaml file is not referred by the .pro file.

Reply:
In that case, wouldn't we already return from either !isReallyActive() or project() checks?

Reply:
Nope. In this context `this` is `MerBuildConfiguration`, not `IDocument` - I guess that is what confused you.